portal-user-models/src/main/java/com/telecominfraproject/wlan/portaluser/models/PortalUser.java

```diff
@@ -79,18 +86,29 @@ public void setPassword(String password) {
 		this.password = password;
 	}
 
-	public PortalUserRole getRole() {
-		return role;
+	public List<PortalUserRole> getRoles() {
+		return roles;
 	}
 
-	public void setRole(PortalUserRole role) {
-		this.role = role;
+	public void setRoles(List<PortalUserRole> roles) {
+		this.roles = roles;
 	}
 
-	public void setDetails(PortalUserDetails details) {
-		this.details = details;
+	@JsonIgnore
+	@Deprecated
+	public PortalUserRole getRole() {
+		if (roles == null || roles.isEmpty()) {
+			return null;
+		}
+		return roles.get(0);
 	}
-
+	
+	@JsonIgnore
+	@Deprecated
+	public void setRole(PortalUserRole role) {
+		roles = Arrays.asList(role);
+	}
```

Comment:
I've been thinking about UI changes required to support the list of roles in cloudSDK. Until UI is changed to work with portalUser.roles property we will need to make sure the portalUser.role works seamlessly.

Here's what I came up with to handle this case:
`
public List<PortalUserRole> getRoles() {
	return roles;
}

public void setRoles(List<PortalUserRole> roles) {
  //This is to cover the backwards-compatibility case when the 
  //roles property is not provided, but the role property is provided.
  //This check can be removed when deprecated methods getRole/setRole are removed.
  if(roles != null){
	 this.roles = roles;
  }
}

@Deprecated
public PortalUserRole getRole() {
  if (roles == null || roles.isEmpty()) {
    return null;
  }
  return roles.get(0);
}

@Deprecated
public void setRole(PortalUserRole role) {
  if(roles == null){
    roles = Arrays.asList(role);
  }
}

`